# Spark Fund Analysis

**Load Libraries**

In [1]:
# Supress Warnings

import warnings
warnings.filterwarnings('ignore')

In [2]:
import numpy as np
import pandas as pd

**Load Investment Data**

In [9]:
import chardet
f = open("companies.txt","rb").read()

In [11]:
result= chardet.detect(f)

In [14]:
result

{'confidence': 0.461086087463078,
 'encoding': 'Windows-1254',
 'language': 'Turkish'}

In [13]:
companies = pd.read_csv("companies.txt", sep="\t", encoding="ISO-8859-1")

UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 17: character maps to <undefined>

In [4]:
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [5]:
companies.tail()

permalink  \
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name            homepage_url  \
66363                  ZZNode Science and Technology   http://www.zznode.com   
66364                          Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
66365                                         ÁERON    http://www.aeron.hu/   
66366                                         Ôasys    http://www.oasys.io/   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic    http://inovatiff.com   

                                           category_list     status  \
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code       region           city  founded_at  
66363          CHN         22      Beijing        Beijing         NaN  
66364          HRV         15        Split          Split  13-05-2012  
66365          NaN        NaN          NaN            NaN  01-01-2011  
66366          USA         CA  SF Bay Area  San Francisco  01-01-2014  
66367          NaN        NaN          NaN            NaN         NaN

Updating unicode character data by assigning correct encoding type.

In [6]:
companies.permalink = companies.permalink.str.encode('utf-8').str.decode('ascii', 'ignore')
companies.name = companies.name.str.encode('utf-8').str.decode('ascii', 'ignore')

In [7]:
companies.tail()

permalink  \
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                                 /Organization/Eron   
66366                               /Organization/Asys-2   
66367  /Organization/Novatiff-Reklam-Ve-TanTM-Hizmetl...   

                                          name            homepage_url  \
66363            ZZNode Science and Technology   http://www.zznode.com   
66364                    Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
66365                                     ERON    http://www.aeron.hu/   
66366                                     asys    http://www.oasys.io/   
66367  novatiff Reklam ve Tantm Hizmetleri Tic    http://inovatiff.com   

                                           category_list     status  \
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code       region           city  founded_at  
66363          CHN         22      Beijing        Beijing         NaN  
66364          HRV         15        Split          Split  13-05-2012  
66365          NaN        NaN          NaN            NaN  01-01-2011  
66366          USA         CA  SF Bay Area  San Francisco  01-01-2014  
66367          NaN        NaN          NaN            NaN         NaN

In [8]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [9]:
print(companies.shape)
companies.describe()

(66368, 10)


permalink   name                homepage_url  \
count                           66368  66367                       61310   
unique                          66368  66102                       61191   
top     /Organization/Milo-Technology  Karma  http://www.askforoffer.com   
freq                                1      4                           5   

       category_list     status country_code state_code       region  \
count          63220      66368        59410      57821        58338   
unique         27296          4          137        311         1092   
top         Software  operating          USA         CA  SF Bay Area   
freq            3995      53034        37601      12900         8804   

                 city  founded_at  
count           58340       51147  
unique           5111        3978  
top     San Francisco  01-01-2012  
freq             3526        2730

As per above result **permlink** column having all unique values. <br />
So, we can use **permalink** as index column

In [10]:
#companies.set_index("permalink", inplace=True)

In [11]:
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [12]:
rounds2 = pd.read_csv("rounds2.csv", encoding="ISO-8859-1")
rounds2.company_permalink = rounds2.company_permalink.str.encode('utf-8').str.decode('ascii', 'ignore')

In [13]:
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [14]:
rounds2.tail()

company_permalink  \
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                                 /organization/eron   
114947                               /ORGANIZATION/ASYS-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink funding_round_type  \
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0

In [15]:
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


In [16]:
rounds2.describe()

raised_amount_usd
count       9.495900e+04
mean        1.042687e+07
std         1.148212e+08
min         0.000000e+00
25%         3.225000e+05
50%         1.680511e+06
75%         7.000000e+06
max         2.127194e+10

In [17]:
rounds2[rounds2.columns[:-1]].describe()

company_permalink  \
count                     114949   
unique                     90247   
top     /ORGANIZATION/SOLARFLARE   
freq                          10   

                                funding_round_permalink funding_round_type  \
count                                            114949             114949   
unique                                           114949                 14   
top     /funding-round/081637fdae7e4b0978f800c1f5cf6c14            venture   
freq                                                  1              55494   

       funding_round_code   funded_at  
count               31140      114949  
unique                  8        5033  
top                     A  01-01-2014  
freq                14747         779

In [18]:
companies["permalink"] = companies["permalink"].apply(lambda x : x.lower())
rounds2["company_permalink"] = rounds2["company_permalink"].apply(lambda x : x.lower())
rounds2["funding_round_permalink"] = rounds2["funding_round_permalink"].apply(lambda x : x.lower())

# Checkpoint 1: Data Cleaning 1

In [19]:
print ("Unique companies in rounds2 data frame:" , rounds2.company_permalink.unique().size)

Unique companies in rounds2 data frame: 66368


In [20]:
print ("Unique companies in companies data frame:" , companies.permalink.unique().size)

Unique companies in companies data frame: 66368


In [21]:
companies.describe()

permalink   name                homepage_url  \
count                     66368  66367                       61310   
unique                    66368  66102                       61191   
top     /organization/cartsmart  Karma  http://www.askforoffer.com   
freq                          1      4                           5   

       category_list     status country_code state_code       region  \
count          63220      66368        59410      57821        58338   
unique         27296          4          137        311         1092   
top         Software  operating          USA         CA  SF Bay Area   
freq            3995      53034        37601      12900         8804   

                 city  founded_at  
count           58340       51147  
unique           5111        3978  
top     San Francisco  01-01-2012  
freq             3526        2730

Companies data frame having total ** 66368 ** rows <br>
and ** permalink ** column having ** 66368 ** unique values. <br>
So we can use ** permalink ** column as unique identifier for row or can use as index column.

In [22]:
master_frame= rounds2.merge(companies, how='left', left_on="company_permalink", right_on="permalink")
master_frame.head()

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
1                  A  14-10-2014                NaN   
2                NaN  01-03-2014           700000.0   
3                  B  30-01-2014          3406878.0   
4                  A  19-03-2008          2000000.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [23]:
print ("Companies only available in rounds2 :",master_frame["permalink"].isnull().sum())

Companies only available in rounds2 : 0


In [24]:
print("Master Frame Detail : ", master_frame.shape)

Master Frame Detail :  (114949, 16)


** Data Cleaning **

In [25]:
df_master = master_frame.copy(deep=True)

Remove unwanted Column

In [26]:
master_frame.drop(["homepage_url", "permalink", "funding_round_code", "founded_at", "funding_round_permalink"], axis=1, inplace=True)
master_frame.shape

(114949, 11)

Remove Null value rows

In [27]:
master_frame.isnull().sum() / len(master_frame.index) * 100

company_permalink      0.000000
funding_round_type     0.000000
funded_at              0.000000
raised_amount_usd     17.390321
name                   0.000870
category_list          2.966533
status                 0.000000
country_code           7.549435
state_code             9.522484
region                 8.844792
city                   8.842182
dtype: float64

In [28]:
master_frame.dropna(how='all', axis = 0, subset=['raised_amount_usd'], inplace=True)
master_frame.dropna(how='all', axis = 0, subset=['category_list'], inplace=True)
master_frame.dropna(how='all', axis = 0, subset=['country_code'], inplace=True)

In [29]:
master_frame.isnull().sum() / len(master_frame.index) * 100

company_permalink     0.000000
funding_round_type    0.000000
funded_at             0.000000
raised_amount_usd     0.000000
name                  0.001130
category_list         0.000000
status                0.000000
country_code          0.000000
state_code            1.955291
region                1.337415
city                  1.334026
dtype: float64

In [30]:
master_frame.shape

(88529, 11)

# Checkpoint 2: Funding Type Analysis

In [31]:
master_frame["raised_amount_million_usd"] = master_frame["raised_amount_usd"] / 1000000

In [32]:
fundTypeGroup = master_frame.groupby(by="funding_round_type")

In [33]:
avgInvestment = fundTypeGroup.mean().loc[['venture','angel','seed','private_equity']]
avgInvestment

raised_amount_usd  raised_amount_million_usd
funding_round_type                                              
venture                  1.172422e+07                  11.724223
angel                    9.715739e+05                   0.971574
seed                     7.477937e+05                   0.747794
private_equity           7.393849e+07                  73.938486

In [34]:
print('Average funding amount of venture type :', avgInvestment["raised_amount_million_usd"].loc['venture'])
print('Average funding amount of angel type :', avgInvestment["raised_amount_million_usd"].loc['angel'])
print('Average funding amount of seed type :', avgInvestment["raised_amount_million_usd"].loc['seed'])
print('Average funding amount of private equity type :', avgInvestment["raised_amount_million_usd"].loc['private_equity'])

Average funding amount of venture type : 11.7242226902
Average funding amount of angel type : 0.971573891136
Average funding amount of seed type : 0.747793682484
Average funding amount of private equity type : 73.9384862753


In [35]:
FT = avgInvestment[(avgInvestment["raised_amount_million_usd"] >= 5) & (avgInvestment["raised_amount_million_usd"] <= 15)].index[0]
print("Investment type suitable for spark fund (5 - 15 Million USD) : ", FT )

Investment type suitable for spark fund (5 - 15 Million USD) :  venture


<font size='4'> ** "Venture" ** is the suitable investment type for Spark Fund </font>

# Checkpoint 3: Country Analysis

** Loading Counties where English is official language **

Using **Contries.csv** file which is created using PDF file from <br>
http://www.emmir.org/fileadmin/user_upload/admission/Countries_where_English_is_an_official_language.pdf

In [36]:
country = pd.read_csv("Countries.csv")
country.head()

Country Code
0    Botswana  BWA
1    Cameroon  CMR
2    Ethiopia  ETH
3     Eritrea  ERI
4  The Gambia  GMB

In [37]:
ventureInventment = master_frame[master_frame["funding_round_type"] == "venture"]
ventureInventment = ventureInventment.merge(country, how="left", left_on="country_code", right_on="Code")
ventureInventment.head()

company_permalink funding_round_type   funded_at  \
0                  /organization/-fame            venture  05-01-2015   
1                /organization/0-6-com            venture  19-03-2008   
2  /organization/0ndine-biomedical-inc            venture  21-12-2009   
3                 /organization/0xdata            venture  09-11-2015   
4                 /organization/0xdata            venture  03-01-2013   

   raised_amount_usd                    name  category_list     status  \
0         10000000.0                   #fame          Media  operating   
1          2000000.0                 0-6.com    Curated Web  operating   
2           719491.0  Ondine Biomedical Inc.  Biotechnology  operating   
3         20000000.0                  H2O.ai      Analytics  operating   
4          1700000.0                  H2O.ai      Analytics  operating   

  country_code state_code       region           city  \
0          IND         16       Mumbai         Mumbai   
1          CHN         22      Beijing        Beijing   
2          CAN         BC    Vancouver      Vancouver   
3          USA         CA  SF Bay Area  Mountain View   
4          USA         CA  SF Bay Area  Mountain View   

   raised_amount_million_usd        Country Code  
0                  10.000000          India  IND  
1                   2.000000            NaN  NaN  
2                   0.719491         Canada  CAN  
3                  20.000000  United States  USA  
4                   1.700000  United States  USA

In [38]:
ventureInventment["Is_Eng_Official_Lang"] = ventureInventment["Code"] == ventureInventment["country_code"]
ventureInventment.head()

company_permalink funding_round_type   funded_at  \
0                  /organization/-fame            venture  05-01-2015   
1                /organization/0-6-com            venture  19-03-2008   
2  /organization/0ndine-biomedical-inc            venture  21-12-2009   
3                 /organization/0xdata            venture  09-11-2015   
4                 /organization/0xdata            venture  03-01-2013   

   raised_amount_usd                    name  category_list     status  \
0         10000000.0                   #fame          Media  operating   
1          2000000.0                 0-6.com    Curated Web  operating   
2           719491.0  Ondine Biomedical Inc.  Biotechnology  operating   
3         20000000.0                  H2O.ai      Analytics  operating   
4          1700000.0                  H2O.ai      Analytics  operating   

  country_code state_code       region           city  \
0          IND         16       Mumbai         Mumbai   
1          CHN         22      Beijing        Beijing   
2          CAN         BC    Vancouver      Vancouver   
3          USA         CA  SF Bay Area  Mountain View   
4          USA         CA  SF Bay Area  Mountain View   

   raised_amount_million_usd        Country Code  Is_Eng_Official_Lang  
0                  10.000000          India  IND                  True  
1                   2.000000            NaN  NaN                 False  
2                   0.719491         Canada  CAN                  True  
3                  20.000000  United States  USA                  True  
4                   1.700000  United States  USA                  True

** Finding top 9 countries where most investments done **

In [39]:
top9 = ventureInventment.groupby(by="country_code").sum()[["raised_amount_million_usd","Is_Eng_Official_Lang"]].sort_values(by="raised_amount_million_usd", ascending=False)[:9]
top9["Is_Eng_Official_Lang"] = top9["Is_Eng_Official_Lang"] > 1
top9

raised_amount_million_usd  Is_Eng_Official_Lang
country_code                                                 
USA                       420068.029342                  True
CHN                        39338.918773                 False
GBR                        20072.813004                  True
IND                        14261.508718                  True
CAN                         9482.217668                  True
FRA                         7226.851352                 False
ISR                         6854.350477                 False
DEU                         6306.921981                 False
JPN                         3167.647127                 False

In [40]:
top9["Rank"] = top9[["raised_amount_million_usd"]][top9["Is_Eng_Official_Lang"] == True].rank(method="max", ascending=False)
top9[top9["Is_Eng_Official_Lang"] == True]

raised_amount_million_usd  Is_Eng_Official_Lang  Rank
country_code                                                       
USA                       420068.029342                  True   1.0
GBR                        20072.813004                  True   2.0
IND                        14261.508718                  True   3.0
CAN                         9482.217668                  True   4.0

In [41]:
Country1 = top9[top9["Rank"] == 1].index[0]
Country2 = top9[top9["Rank"] == 2].index[0]
Country3 = top9[top9["Rank"] == 3].index[0]

print("Top English-speaking country with most investment : ", Country1)
print("Second English-speaking country with most investment : ", Country2)
print("Third English-speaking country with most investment : ", Country3)

Top English-speaking country with most investment :  USA
Second English-speaking country with most investment :  GBR
Third English-speaking country with most investment :  IND


# Checkpoint 4: Sector Analysis 1

In [42]:
mapping = pd.read_csv("mapping.csv")

In [43]:
print(mapping.shape)
mapping.head()

(688, 10)


category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

In [44]:
mapping.dropna(subset=["category_list"], inplace=True)
print(mapping.shape)
mapping.head()

(687, 10)


category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
1                3D                    0       0                           0   
2       3D Printing                    0       0                           0   
3     3D Technology                    0       0                           0   
4        Accounting                    0       0                           0   
5  Active Lifestyle                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   
5              0       1              0                           0       0   

   Social, Finance, Analytics, Advertising  
1                                        0  
2                                        0  
3                                        0  
4                                        1  
5                                        0

In [45]:
mapping.iloc[[21]]

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
22      A0lytics                    0       0                           0   

    Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
22              0       0              0                           0       0   

    Social, Finance, Analytics, Advertising  
22                                        1

** Cleaning & Transforming Mapping data, so that can be used for deriving Sector **

Few of the category names are corrupted,<br>
Eg. category name ** A0lytics **, here the correct name should be ** Analytics **, so we have to remove **0** from such category names. 

In [46]:
import re
mapping["category_list"] = mapping["category_list"].apply(lambda s : re.sub(r'(\w[0])*0(\s|\w+)', r'\1na\2',s).lower())
mapping.iloc[[21]]

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
22     analytics                    0       0                           0   

    Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
22              0       0              0                           0       0   

    Social, Finance, Analytics, Advertising  
22                                        1

In [47]:
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
1                3d                    0       0                           0   
2       3d printing                    0       0                           0   
3     3d technology                    0       0                           0   
4        accounting                    0       0                           0   
5  active lifestyle                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   
5              0       1              0                           0       0   

   Social, Finance, Analytics, Advertising  
1                                        0  
2                                        0  
3                                        0  
4                                        1  
5                                        0

Derive Main Sector name innew column for all category list

In [48]:
#mapping["main_sector"] = mapping.apply(lambda x : (mapping.loc[x.name] == 1).argmax(), axis = 1);
sectorMapping = pd.melt(mapping, id_vars=['category_list'], var_name='main_sector')
sectorMapping = sectorMapping[sectorMapping["value"] == 1]

mapping = mapping.merge(sectorMapping[["category_list","main_sector"]], on="category_list")
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0                3d                    0       0                           0   
1       3d printing                    0       0                           0   
2     3d technology                    0       0                           0   
3        accounting                    0       0                           0   
4  active lifestyle                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              1                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              0                           0       0   
4              0       1              0                           0       0   

   Social, Finance, Analytics, Advertising  \
0                                        0   
1                                        0   
2                                        0   
3                                        1   
4                                        0   

                               main_sector  
0                            Manufacturing  
1                            Manufacturing  
2                            Manufacturing  
3  Social, Finance, Analytics, Advertising  
4                                   Health

In [49]:
master_frame["primary_sector"] = master_frame["category_list"].apply(lambda x : x.split('|')[0].lower())

In [50]:
master_frame = master_frame.merge(mapping[['category_list', 'main_sector']], how="left", left_on="primary_sector", right_on="category_list", )
master_frame = master_frame.drop(["category_list_y"], axis=1)
master_frame.head()

company_permalink funding_round_type   funded_at  \
0                  /organization/-fame            venture  05-01-2015   
1               /organization/-qounter               seed  01-03-2014   
2                /organization/0-6-com            venture  19-03-2008   
3     /organization/01games-technology        undisclosed  01-07-2014   
4  /organization/0ndine-biomedical-inc               seed  11-09-2009   

   raised_amount_usd                    name  \
0         10000000.0                   #fame   
1           700000.0                :Qounter   
2          2000000.0                 0-6.com   
3            41250.0      01Games Technology   
4            43360.0  Ondine Biomedical Inc.   

                                     category_list_x     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                        Curated Web  operating          CHN   
3                                              Games  operating          HKG   
4                                      Biotechnology  operating          CAN   

  state_code      region           city  raised_amount_million_usd  \
0         16      Mumbai         Mumbai                   10.00000   
1         DE  DE - Other  Delaware City                    0.70000   
2         22     Beijing        Beijing                    2.00000   
3        NaN   Hong Kong      Hong Kong                    0.04125   
4         BC   Vancouver      Vancouver                    0.04336   

          primary_sector                 main_sector  
0                  media               Entertainment  
1  application platforms  News, Search and Messaging  
2            curated web  News, Search and Messaging  
3                  games               Entertainment  
4          biotechnology  Cleantech / Semiconductors

In [51]:
print("Null values for main sector : ", master_frame["main_sector"].isnull().sum() / master_frame.index.size * 100)

Null values for main sector :  0.0711631216889


** We cannot perform sectorwise analysis on null sector values **

In [52]:
master_frame.dropna(subset=["main_sector"], axis = 0,inplace=True)

# Checkpoint 5: Sector Analysis 2

Create Data Frame for **Top 3 English Speaking** countries

In [53]:
standFilter =  (master_frame["raised_amount_million_usd"] >= 5) & (master_frame["raised_amount_million_usd"] <= 15) & (master_frame["funding_round_type"] == FT)
D1 = master_frame[(master_frame["country_code"] == Country1) & standFilter]
D2 = master_frame[(master_frame["country_code"] == Country2) & standFilter]
D3 = master_frame[(master_frame["country_code"] == Country3) & standFilter]

In [54]:
print(D1.shape)
D1.head()

(12063, 14)


company_permalink funding_round_type   funded_at  \
9              /organization/0xdata            venture  19-07-2014   
15       /organization/1-mainstream            venture  17-03-2015   
71     /organization/128-technology            venture  16-12-2014   
77  /organization/1366-technologies            venture  15-10-2013   
78  /organization/1366-technologies            venture  09-04-2015   

    raised_amount_usd               name                   category_list_x  \
9           8900000.0             H2O.ai                         Analytics   
15          5000000.0       1 Mainstream  Apps|Cable|Distribution|Software   
71         11999347.0     128 Technology      Service Providers|Technology   
77         15000000.0  1366 Technologies                     Manufacturing   
78          5000000.0  1366 Technologies                     Manufacturing   

       status country_code state_code       region           city  \
9   operating          USA         CA  SF Bay Area  Mountain View   
15   acquired          USA         CA  SF Bay Area      Cupertino   
71  operating          USA         MA       Boston     Burlington   
77  operating          USA         MA       Boston        Bedford   
78  operating          USA         MA       Boston        Bedford   

    raised_amount_million_usd     primary_sector  \
9                    8.900000          analytics   
15                   5.000000               apps   
71                  11.999347  service providers   
77                  15.000000      manufacturing   
78                   5.000000      manufacturing   

                                main_sector  
9   Social, Finance, Analytics, Advertising  
15               News, Search and Messaging  
71                                   Others  
77                            Manufacturing  
78                            Manufacturing

In [55]:
print(D2.shape)
D2.head()

(621, 14)


company_permalink funding_round_type   funded_at  \
289  /organization/365scores            venture  29-09-2014   
623   /organization/7digital            venture  01-01-2008   
624   /organization/7digital            venture  19-10-2012   
677      /organization/90min            venture  06-10-2015   
678      /organization/90min            venture  07-05-2013   

     raised_amount_usd       name  \
289          5500000.0  365Scores   
623          8468328.0   7digital   
624         10000000.0   7digital   
677         15000000.0      90min   
678          5800000.0      90min   

                                       category_list_x     status  \
289                  Android|Apps|iPhone|Mobile|Sports  operating   
623  Content Creators|Content Delivery|Licensing|Mu...   acquired   
624  Content Creators|Content Delivery|Licensing|Mu...   acquired   
677                Media|News|Publishing|Soccer|Sports  operating   
678                Media|News|Publishing|Soccer|Sports  operating   

    country_code state_code  region    city  raised_amount_million_usd  \
289          GBR         H9  London  London                   5.500000   
623          GBR         H9  London  London                   8.468328   
624          GBR         H9  London  London                  10.000000   
677          GBR         H9  London  London                  15.000000   
678          GBR         H9  London  London                   5.800000   

       primary_sector                              main_sector  
289           android  Social, Finance, Analytics, Advertising  
623  content creators                            Entertainment  
624  content creators                            Entertainment  
677             media                            Entertainment  
678             media                            Entertainment

In [56]:
print(D3.shape)
D3.head()

(328, 14)


company_permalink funding_round_type   funded_at  \
0                /organization/-fame            venture  05-01-2015   
162   /organization/21diamonds-india            venture  15-11-2012   
753     /organization/a-little-world            venture  09-09-2008   
1920    /organization/adlabs-imagica            venture  28-10-2014   
2665             /organization/agile            venture  01-05-2011   

      raised_amount_usd            name  \
0            10000000.0           #fame   
162           6369507.0      21Diamonds   
753           6410000.0  A LITTLE WORLD   
1920          8180000.0  Adlabs Imagica   
2665          5740000.0           Agile   

                                   category_list_x     status country_code  \
0                                            Media  operating          IND   
162                                     E-Commerce  operating          IND   
753                                        Finance  operating          IND   
1920                         Entertainment|Tourism  operating          IND   
2665  Finance|Finance Technology|FinTech|Insurance  operating          IND   

     state_code       region     city  raised_amount_million_usd  \
0            16       Mumbai   Mumbai                  10.000000   
162          10    New Delhi  Gurgaon                   6.369507   
753          16       Mumbai   Mumbai                   6.410000   
1920         16  IND - Other  Khopoli                   8.180000   
2665         16       Mumbai   Mumbai                   5.740000   

     primary_sector                              main_sector  
0             media                            Entertainment  
162      e-commerce                                   Others  
753         finance  Social, Finance, Analytics, Advertising  
1920  entertainment                            Entertainment  
2665        finance  Social, Finance, Analytics, Advertising

** Creating Sectorwise groups **

In [57]:
d1SectorGroupC = D1.groupby(by="main_sector")["raised_amount_million_usd"].count()
d1SectorGroupS = D1.groupby(by="main_sector")["raised_amount_million_usd"].sum()

D1["Total_Num_Investments"] = D1["main_sector"].apply(lambda s : d1SectorGroupC[s])
D1["Total_AmountInvested"] = D1["main_sector"].apply(lambda s : d1SectorGroupS[s])

In [58]:
d2SectorGroupC = D2.groupby(by="main_sector")["raised_amount_million_usd"].count()
d2SectorGroupS = D2.groupby(by="main_sector")["raised_amount_million_usd"].sum()

D2["Total_Num_Investments"] = D2["main_sector"].apply(lambda s : d2SectorGroupC[s])
D2["Total_AmountInvested"] = D2["main_sector"].apply(lambda s : d2SectorGroupS[s])

In [59]:
d3SectorGroupC = D3.groupby(by="main_sector")["raised_amount_million_usd"].count()
d3SectorGroupS = D3.groupby(by="main_sector")["raised_amount_million_usd"].sum()

D3["Total_Num_Investments"] = D3["main_sector"].apply(lambda s : d3SectorGroupC[s])
D3["Total_AmountInvested"] = D3["main_sector"].apply(lambda s : d3SectorGroupS[s])

** Table 5.1 : Sector-wise Investment Analysis ** <br>
&nbsp; &nbsp; 1. Total number of investments <br>
&nbsp; &nbsp; 2. Total amount of investment

In [60]:
print("Total Number Of Investment in Country 1 : ", D1.index.size)
print("Total Number Of Investment in Country 2 : ", D2.index.size)
print("Total Number Of Investment in Country 3 : ", D3.index.size)

print("Total Amount Invested in Country 1 : ", D1["raised_amount_usd"].sum())
print("Total Amount Invested in Country 2 : ", D2["raised_amount_usd"].sum())
print("Total Amount Invested in Country 3 : ", D3["raised_amount_usd"].sum())

Total Number Of Investment in Country 1 :  12063
Total Number Of Investment in Country 2 :  621
Total Number Of Investment in Country 3 :  328
Total Amount Invested in Country 1 :  107757097294.0
Total Amount Invested in Country 2 :  5379078691.0
Total Amount Invested in Country 3 :  2949543602.0


&nbsp; &nbsp; 3. Top sector (based on count of investments)<br>
&nbsp; &nbsp; 4. Second-best sector (based on count of investments)<br>
&nbsp; &nbsp; 5. Third-best sector (based on count of investments)<br>
&nbsp; &nbsp; 6. Number of investments in the top sector (refer to point 3)<br>
&nbsp; &nbsp; 7. Number of investments in the second-best sector (refer to point 4)<br>
&nbsp; &nbsp; 8. Number of investments in the third-best sector (refer to point 5)

In [61]:
print("Country 1 ({0}) : Sectorwise Top 3 Investment Count ".format(Country1))
print(d1SectorGroupC.sort_values(ascending=False)[:3])

Country 1 (USA) : Sectorwise Top 3 Investment Count 
main_sector
Others                                     2950
Social, Finance, Analytics, Advertising    2714
Cleantech / Semiconductors                 2350
Name: raised_amount_million_usd, dtype: int64


In [62]:
print("Country 2 ({0}) : Sectorwise Top 3 Investment Count ".format(Country2))
print(d2SectorGroupC.sort_values(ascending=False)[:3])

Country 2 (GBR) : Sectorwise Top 3 Investment Count 
main_sector
Others                                     147
Social, Finance, Analytics, Advertising    133
Cleantech / Semiconductors                 130
Name: raised_amount_million_usd, dtype: int64


In [63]:
print("Country 3 ({0}) : Sectorwise Top 3 Investment Count ".format(Country3))
print(d3SectorGroupC.sort_values(ascending=False)[:3])

Country 3 (IND) : Sectorwise Top 3 Investment Count 
main_sector
Others                                     110
Social, Finance, Analytics, Advertising     60
News, Search and Messaging                  52
Name: raised_amount_million_usd, dtype: int64


&nbsp; &nbsp; 9. For the top sector count-wise (point 3), which company received the highest investment? <br>
&nbsp; &nbsp; 10. For the second-best sector count-wise (point 4), which company received the highest investment?

In [64]:
print("Country 1 ({0}) : Highest Investement Recevied in 'Other' sector by Company :".format(Country1))
D1[D1["main_sector"] == "Others"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 1 (USA) : Highest Investement Recevied in 'Other' sector by Company :


raised_amount_usd  raised_amount_million_usd  \
company_permalink                                                         
/organization/virtustream         64300000.0                       64.3   

                           Total_Num_Investments  Total_AmountInvested  
company_permalink                                                       
/organization/virtustream                  17700         157926.042012

In [65]:
print("Country 1 ({0}) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :".format(Country1))
D1[D1["main_sector"] == "Social, Finance, Analytics, Advertising"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 1 (USA) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :


raised_amount_usd  raised_amount_million_usd  \
company_permalink                                                         
/organization/shotspotter         67933006.0                  67.933006   

                           Total_Num_Investments  Total_AmountInvested  
company_permalink                                                       
/organization/shotspotter                  18998         166651.638748

In [66]:
print("Country 2 ({0}) : Highest Investement Recevied in 'Other' sector by Company :".format(Country2))
D2[D2["main_sector"] == "Others"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 2 (GBR) : Highest Investement Recevied in 'Other' sector by Company :


raised_amount_usd  raised_amount_million_usd  \
company_permalink                                                            
/organization/electric-cloud         37000000.0                       37.0   

                              Total_Num_Investments  Total_AmountInvested  
company_permalink                                                          
/organization/electric-cloud                    588           5134.497156

In [67]:
print("Country 2 ({0}) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :".format(Country2))
D2[D2["main_sector"] == "Social, Finance, Analytics, Advertising"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 2 (GBR) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :


raised_amount_usd  \
company_permalink                                        
/organization/celltick-technologies         37500000.0   

                                     raised_amount_million_usd  \
company_permalink                                                
/organization/celltick-technologies                       37.5   

                                     Total_Num_Investments  \
company_permalink                                            
/organization/celltick-technologies                    532   

                                     Total_AmountInvested  
company_permalink                                          
/organization/celltick-technologies           4357.616056

In [68]:
print("Country 3 ({0}) : Highest Investement Recevied in 'Other' sector by Company :".format(Country3))
D3[D3["main_sector"] == "Others"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 3 (IND) : Highest Investement Recevied in 'Other' sector by Company :


raised_amount_usd  raised_amount_million_usd  \
company_permalink                                                          
/organization/firstcry-com         39000000.0                       39.0   

                            Total_Num_Investments  Total_AmountInvested  
company_permalink                                                        
/organization/firstcry-com                    330           3040.228521

In [69]:
print("Country 3 ({0}) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :".format(Country3))
D3[D3["main_sector"] == "Social, Finance, Analytics, Advertising"].groupby(by="company_permalink").sum().sort_values(by="raised_amount_million_usd", ascending=False)[:1]

Country 3 (IND) : Highest Investement Recevied in 'Social, Finance, Analytics, Advertising' sector by Company :


raised_amount_usd  raised_amount_million_usd  \
company_permalink                                                             
/organization/manthan-systems         50700000.0                       50.7   

                               Total_Num_Investments  Total_AmountInvested  
company_permalink                                                           
/organization/manthan-systems                    240             2202.1982

# Checkpoint 6: Plots

** Export Data to excel to use in Tableau for further analysis **

In [70]:
master_frame.to_excel("Master_Frame.xlsx", index=False)

In [71]:
top9.to_excel("Top9.xlsx")